### PR TITLE
Hardlight Bows

### PR DESCRIPTION
**Pull request by silicons**
Adds hardlight bows and hardlight quivers.
Does EDIT: 12 brute and 18 burn on hit.
The quivers regenerate one arrow every few seconds and is worn on the back.
Arrows disintegrate on hit and are destroyed after 20 seconds if it is dropped to prevent spam.
http://imgur.com/a/932lN 
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=8163&sid=12160c77258df6679a0f36e9ce38d692#p217519
Sprites by the amazing @Lexorion 
ADDED TO UPLINK AT 7 TELECRYSTALS FOR A BOW AND QUIVER. Can be included in surplus crates.
:cl:
rscadd: Adds a hardlight bow and hardlight quiver that do 12 brute and 18 burn on impact, and regenerates an arrow every few seconds for the quiver.
experimental: Syndicate personnel may buy this bundle at the low low price of 7 telecrystals.
:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
GET ON THEME

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 1]
```diff
@@ -29,10 +38,12 @@
 
 /obj/item/weapon/gun/projectile/bow/attack_self(mob/living/user)
 	if(!ready_to_fire && magazine.ammo_count())
+		spawn(draw_time)
```
Comment (AnturK) [inline code/modules/projectiles/guns/projectile/bow.dm]: This does nothing here so probably not what you intended to do. Ideally use do_after here so the bar shows up.

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 2]
```diff
@@ -29,10 +38,12 @@
...
 		ready_to_fire = TRUE
 		playsound(user, draw_sound, 100, 1)
 		update_icon()
 	else
+		spawn(draw_time)
```
Comment (AnturK) [inline code/modules/projectiles/guns/projectile/bow.dm]: Same as above.

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 3]
```diff
@@ -96,3 +107,85 @@
...
+			new /obj/item/ammo_casing/caseless/arrow/hardlight(src)
+		charge_tick = 0
+
+/obj/item/projectile/bullet/reusable/arrow/hardlight/on_hit(atom/target, blocked = 0)
+	..()
+	var/obj/item/projectile/bullet/reusable/arrow/hardlight2/A = new /obj/item/projectile/bullet/reusable/arrow/hardlight2(src.loc)
+	A.Bump(target, 1)
+
+/obj/item/ammo_casing/caseless/arrow/hardlight/Dropped()
+	addtimer(src, "Destroy", 200)
```
Comment (AnturK) [inline code/modules/projectiles/guns/projectile/bow.dm]: We have a helper for this. qdel_in(src,200)

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by Lexorion** [comment]
tw1yrcm
Pic from the forum

File: code/modules/projectiles/guns/projectile/bow.dm [hunk 4]
```diff
@@ -96,3 +107,85 @@
 	..()
 	for(var/i in 1 to storage_slots)
 		new /obj/item/ammo_casing/caseless/arrow(src)
+
+/obj/item/weapon/storage/backpack/quiver/proc/arrow_count()
+	var/count = 0
+	for(var/obj/item/ammo_casing/caseless/arrow/A in src)
+		count++
```
Comment (kevinz000) [inline code/modules/projectiles/guns/projectile/bow.dm]: Uh, for quivers as a whole? Because the normal one has no need to track arrow amounts

Comment (phil235) [inline code/modules/projectiles/guns/projectile/bow.dm]: make it use a var, we're not going to do an istype on every single arrow in the quiver to count them, it's tedious.

Comment (Iamgoofball) [inline code/modules/projectiles/guns/projectile/bow.dm]: or contents.len

**Comment by xxalpha** [comment]
>get on theme

Remove bone spears and medieval bows, first. At least this pretends to be sci-fi.

**Comment by Cheridan** [comment]
This isn't scifi at all you mong, it's horrible sci-fantasy shit like Jupiter Ascending.

**Comment by TechnoAlchemisto** [comment]
hardlight is apex pseudo sci-fi and I fully support it

**Comment by Cheridan** [comment]
sci fantasy

WHEE MR RABBIT LETS RIDE THE SHLOCK BIKE OFF TO WORST-FUCKING-GENRE-EVERVILLE 
WE CAN TALK TO GEORGE FUCKING LUCAS ALONG THE WAY

**Comment by optimumtact** [comment]
I'm with cheridan

**Comment by TechnoAlchemisto** [comment]
how does she breath underwater

**Comment by Cheridan** [comment]
how do you shoot an arrow with a laser?? who cares, it's fantasy!

File: code/modules/projectiles/guns/projectile/bow.dm (hunk 3 above)
Comment (kevinz000) [inline code/modules/projectiles/guns/projectile/bow.dm]: fixed

File: code/modules/projectiles/guns/projectile/bow.dm (hunk 2 above)
Comment (kevinz000) [inline code/modules/projectiles/guns/projectile/bow.dm]: removed

File: code/modules/projectiles/guns/projectile/bow.dm (hunk 1 above)
Comment (kevinz000) [inline code/modules/projectiles/guns/projectile/bow.dm]: removed

**Comment by Lexorion** [comment]
But it's not laser, it's hardlight and cold , in games like Halo you can walk over bridges made out of it without any problems.

**Comment by kevinz000** [comment]
Requesting testmerge. :3

**Comment by Durkel** [comment]
I don't like this, it just seems silly, no one in their right mind is going to use a bow and arrow in the future when you can just pull a trigger. We're also going down a dangerous path if we start including "hardlight" shit, because you can make literally anything out of that and bullshit it off as it being the "future".

**Comment by Lexorion** [comment]
Why "dangerous path"? We already have e-(cross)bows,e-swords and e-shields but a different approach on something quite similar is suddenly bad and 100% not fitting? I mean yes,you *could* make a hardlight variant for that sort of stuff but I can't think of anything else plus they are not that different,e.g. the "energy" shield doesn't burn you/others and is solid so wouldn't that basically make it hardlight or atleast something fairly similar to it?

**Comment by optimumtact** [comment]
We have previous mistakes, but you don't keep crashing your car into things just because you have a few previous dents.

**Comment by kevinz000** [comment]
... that's a really harsh way to put it @optimumtact

**Comment by Bawhoppen** [comment]
only problem with this is that its available for syndicks

**Comment by nullbear** [comment]
I see nothing wrong with the 'hardlight' aspect of this (and in other PR's). It has good gameplay applications. If you want to argue that it's not hard-sci-fi, you can pretty easily pull-shit-out-of-your-ass to explain that away anyways. 

(IE. it is made up of: compressible ferrous semiconductive aerogel particles with electric and electromagnetic properties. IE: The particles can be held in a specific magnetic field, causing it to fluoresce, and making it appear to hold a solid form in unique and useful shapes + it's ultralight + it doesn't take up any space when inactive + it's ultrasharp/forceful + self-healing/immune to damage (since it's made up of particles which are NOT chemically bonded to eachother.))

Having the BOW as a weapon is no better or worse than the pneumatic cannon. It's a gimmick weapon who's only purpose is to fill a role as a weapon through interesting/unique mechanics. Should it be seen frequently in the game? No. It fits much better in the sci-fi role than your uh... medieval bows. And appears to be quite a bit more fleshed out as well. 

I personally see it as kind of the same way as hunters who use compound bows. It's a modernized archaic weapon that is 'worse' than the modern weapon (guns.) but offers an alternative method of hunting which is definitely more interesting, and challenging. IRL: bows are silent, and the projectiles arc. 

The only basis i can see this PR being denied on is on HOW the bow is introduced and made available. Should it be available to syndicates? No. Would it make an interesting ruin/megafauna loot? For sure, and it's better than angel wings or drake blood in regards to the theme to boot.

**Review by AnturK — APPROVED** [review]

**Comment by xxalpha** [comment]
We added caveman bone spears and medieval wooden bows not that long ago, I just want to point out that again.

**Comment by KorPhaeron** [comment]
>spears aren't realistic or scientific for tribal people to be using

?

**Comment by Durkel** [comment]
Bows shouldn't have been added, the pneumatic cannon is a balance nightmare and can crit someone in one hit, Just because dumb shit slips by doesn't mean we should keep adding more. And pointy sticks are always going to be a thing, not even the future can change that. Just because something is coded and can be added to the game doesn't mean it should be, you should ask the questions of : "what will this do", "what will its impact be" and "Do we really need this". 

Traitor items are already bloated with gimmick shit that's rarely picked if not for a surplus crate, Do we really need more?

**Comment by kevinz000** [comment]
good thing this is a viable weapon and not a gimmicky shit right
Oh yeah, should I make this deal full damage to lavaland mobs
Might be useful for lavaland if somehow this doesn't make it into traitor stuff :^)

**Comment by Cheridan** [comment]
Gritty tribal weapons are more on-theme for SS13 than laserstring tekbows.

**Comment by ExcessiveUseOfCobblestone** [comment]
that bunny is pretty cute though cheri...

**Comment by Hatfish** [comment]
Why are demons and magical bullshit allowed but this is somehow going over the line?

**Comment by optimumtact** [comment]
because the source for this particular lore is garbage

**Comment by xxalpha** [comment]
So is Dead Space tbh.

**Comment by KorPhaeron** [comment]
@Hatfish 

A wizard getting a new magical whatever makes sense for the wizard, thats internal consistency and theme for the faction.

The captain suddenly getting a magic wand would be silly.

And similarly, it's a bit odd that the syndicate, which relies on old solid projectile weapons, would have something way more advanced than NT does

**Comment by PKPenguin321** [comment]
Maybe swap out R&D energy crossbow with this?

**Comment by kevinz000** [comment]
^ that'd honestly nerf R&D as this 1. can't stunlock 2. has to be back-worn or inhand-held because it's a large item.

**Comment by PKPenguin321** [comment]
Sounds good, do it

**Comment by optimumtact** [comment]
sounds okay to me

**Comment by Bawhoppen** [comment]
I vote for nerfing R&D while also giving it a more thematic and unique weapon whilst removing a duplicate mechanic weapon simultaneously

**Comment by nullbear** [comment]
Why is this still open geez it's been like a month

**Comment by xxalpha** [comment]
@KorPhaeron can't use that as an argument as we have let security have bullets already.

**Comment by optimumtact** [comment]
Replace the R&D energy crossbow with this and I will pull it.

**Comment by kevinz000** [comment]
Oh
Man
You're driving a hard bargain sir
UHHHh how about if this is attachable to the quiver so you don't have to hold it and wear the quiver and instead you can wear both on your back?

**Comment by PKPenguin321** [comment]
>it's a science item now

>okay uh how about we buff it?

l o l

**Comment by optimumtact** [comment]
OOC: Dutch Blitz: Tell oranges he can have my hardlight

**Comment by kevinz000** [comment]
WTF

**Comment by optimumtact** [comment]
No buffs kevzzzzzzzz it's fine as it is

File: code/modules/research/designs/weapon_designs.dm [hunk 5]
```diff
@@ -83,6 +83,7 @@
 	build_path = /obj/item/weapon/gun/syringe/rapidsyringe
 	category = list("Weapons")
 
+/* Fruits won.
```
Comment (PKPenguin321) [inline code/modules/research/designs/weapon_designs.dm]: ooc in code ban he

**Pull request merged**